rasa/core/actions/action.py

```diff
@@ -735,36 +735,33 @@ async def run(
             return bot_messages + evts
 
         except ClientResponseError as e:
-            if e.status == 400:
-                response_data = json.loads(e.text)
-                exception = ActionExecutionRejection(
-                    response_data["action_name"], response_data.get("error")
-                )
-                logger.error(exception.message)
-                raise exception
-            else:
-                raise RasaException("Failed to execute custom action.") from e
+            if e.status != 400:
+                raise RasaException('Failed to execute custom action.') from e
+            response_data = json.loads(e.text)
+            exception = ActionExecutionRejection(
+                response_data['action_name'], response_data.get('error')
+            )
 
+            logger.error(exception.message)
+            raise exception
```

Comment:
Function `RemoteAction.run` refactored with the following changes:

- Swap if/else branches ([swap-if-else-branches](https://docs.sourcery.ai/refactorings/swap-if-else-branches/))
- Remove unnecessary else after guard condition ([remove-unnecessary-else](https://docs.sourcery.ai/refactorings/remove-unnecessary-else/))